### PR TITLE
Enable production deployment

### LINKED ISSUE #93: Generated files should not be tracked
Thinking specifically of `build.js`.

### LINKED ISSUE #142: Set up a staging server
I think it's time we started thinking just the tiniest bit about devops. Don't want to leave it to the very last minute. I propose we stand up a digital ocean droplet (or equivalent) and work up a set of scripts and nginx configs to get this thing serving.

### PR DESCRIPTION
**Pull request by GabeIsman**
This pull request contains a copy of an nginx config file for a mediapublic server, notes on server setup, and a substantial reworking of the gulp tasks to enable easy deployment of the client.

I have introduced the gulp task 'deploy' which:
1. Cleans out old production files
2. Builds fresh versions main.min.css and main.min.js
3. Compresses the assets
4. Names them uniquely to get around caching issues
5. Rebuilds index.html with the appropriate asset paths
6. Copies the public folder to a remote server

This is all running at http://45.55.211.171/ (DNS coming soon!).

Major bonus in all this is that I've now untracked the generated assets, which should produce a 99% reduction in merge conflicts, hurray!

Resolves #142. Resolves #93.

### CODE REVIEW

**Comment by ryansb** [comment]
:+1: Here, have some DNS http://mediapublic.ryansb.com/

File: production/mediapublic [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+
+# HTTPS server
+#
+#server {
+# listen 443;
+# server_name localhost;
+#
+# root html;
```
Comment (ryansb) [inline production/mediapublic:94]: Can you delete these comments?

File: production/notes.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+create user, add to sudoers, install ssh key
```
Comment (ryansb) [inline production/notes.md:1]: Can we maybe make this a script that actually runs as root?

Comment (GabeIsman) [inline production/notes.md:1]: That would be good. I was just planning to create an image of the droplet
for expediency's sake. But if you want to create a setup script feel free!

File: production/mediapublic (hunk 1 above)
Comment (GabeIsman) [inline production/mediapublic:94]: I was thinking we might want to enable some or all of these features down
the line.

Comment (thequbit) [inline production/mediapublic:94]: I'm pro leaving commented code in the code base when in rapid development mode like this.  Later down the line I do agree we should pull it out.

**Comment by thequbit** [comment]
Merged to move forward.  Will open issues as needed.

**Pull request merged**